### DISCUSSION TITLE
使用graph.fromJSON 渲染数据报错-Error: The `shape` should be specified when creating a node/edge instance

### DISCUSSION THREAD

**Discussion by RockLD** [antvis/X6#3277] [Q&A] [answered]
> @antv/x6版本 2.3.0
> vue 2.x

我在先用graph.toJSON方法导出了数据，节点是html节点，在另一个画布使用graph.fromJSON导入数据的时候报如下错误：
![image](https://user-images.githubusercontent.com/46001500/220571532-4d7d7636-19ec-4cf7-be0c-ce058f729501.png)
代码如下：
![image](https://user-images.githubusercontent.com/46001500/220571878-5d1552ef-88f6-4b54-bfc6-36aa46114674.png)
fromJSON中d的参数为：
```
{
    "cells": [
        {
            "position": {
                "x": 230,
                "y": 130
            },
            "size": {
                "width": 10,
                "height": 10
            },
            "view": "html-view",
            "shape": "red-circle-html",
            "id": "c6015e67-f8d7-43a3-928c-511c10df3944",
            "label": "红色",
            "zIndex": 1
        },
    ]
}
```
有大佬提供一些解决方案或者思路么

**Comment by NewByVector** [comment, marked as answer]
是不是 red-circle-html 这个类型节点还没有注册。

**Reply by RockLD** [reply]
感谢感谢，自定义的节点还需要在当前页面再注册一遍，这回明白了
